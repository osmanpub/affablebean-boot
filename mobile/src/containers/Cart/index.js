import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { ScrollView } from "react-native";
import CartItem from "../../components/CartItem";
import Header from "../../components/Header";

class Cart extends Component {
  render() {
    const { cart, dispatch, setScreen } = this.props;

    const cartItems = cart.items.map((item, index) => (
      <CartItem key={index} dispatch={dispatch} item={item} />
    ));

    return (
      <ScrollView>
        <Header cart={cart} currentScreen="Cart" setScreen={setScreen} />
        {cartItems}
      </ScrollView>
    );
  }
}

const mapStateToProps = state => {
  const { cart } = state;

  return {
    cart
  };
};

export const ConnectedCart = connect(mapStateToProps)(Cart);

Cart.propTypes = {
  cart: PropTypes.object,
  dispatch: PropTypes.func.isRequired,
  setScreen: PropTypes.func.isRequired
};
