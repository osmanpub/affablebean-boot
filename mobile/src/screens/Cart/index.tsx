import React from 'react';
import {ScrollView} from 'react-native';
import {connect} from 'react-redux';
import CartItem from '../../components/CartItem';
import Header from '../../components/Header';
import {Cart as CartState} from '../../interfaces/cart';
import {RootState} from '../../redux';

type Props = {
  cart: CartState;
  setScreen: Function;
};

function Cart(props: Props) {
  const {cart, setScreen} = props;

  if (cart.numberOfItems === 0) {
    setScreen('Home');
    return null;
  }

  const cartItems = cart.items.map((item, index) => (
    <CartItem key={index} item={item} />
  ));

  return (
    <ScrollView>
      <Header currentScreen="Cart" setScreen={setScreen} />
      {cartItems}
    </ScrollView>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart,
});

export default connect(mapStateToProps)(Cart);
