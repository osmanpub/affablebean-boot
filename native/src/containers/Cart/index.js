import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { ScrollView, Text } from "react-native";
import CartItem from "../../components/CartItem";
import Header from "../../components/Header";
// import { clearCart } from "../../actions";

class Cart extends Component {
  componentDidUpdate(prevProps) {
    // const { cart, dispatch, match } = this.props;
    // const { params } = match;
    // if (cart.numberOfItems > 0 && params.clear === "true") {
    //   dispatch(clearCart());
    // }
  }

  render() {
    const { cart, dispatch, setScreen } = this.props;
    const { numberOfItems } = cart;

    const items = cart.items.map((item, index) => (
      <CartItem
        key={item.product.id}
        dispatch={dispatch}
        index={index}
        item={item}
      />
    ));

    // const clearCart =
    //   numberOfItems === 0 ? (
    //     ""
    //   ) : (
    //     <Link className="bubble hMargin" to={"/viewCart/true"}>
    //       clear cart
    //     </Link>
    //   );

    return (
      <ScrollView>
        <Header cart={cart} currentScreen="Cart" setScreen={setScreen} />
      </ScrollView>
    );
    // return (
    //   <div>
    //     <Header cart={cart} url={match.url} />
    //     <div className="singleColumn">
    //       {numberOfItems === 0
    //         ? "Your shopping cart is empty"
    //         : "Your shopping cart contains " + numberOfItems + " items"}
    //       <ActionBar>
    //         {clearCart}
    //         <Link className="bubble hMargin" to={"/"}>
    //           continue shopping
    //         </Link>
    //       </ActionBar>
    //       <ShoppingCart {...cart}>
    //         <Subtotal>
    //           subtotal &euro;
    //           {cart.subtotal.toFixed(2)}
    //         </Subtotal>
    //         <CartTable>
    //           <tbody>
    //             <tr className="header">
    //               <th>product</th>
    //               <th>name</th>
    //               <th>price</th>
    //               <th>quantity</th>
    //             </tr>
    //             {items}
    //           </tbody>
    //         </CartTable>
    //       </ShoppingCart>
    //     </div>
    //     <Footer />
    //   </div>
    // );
  }
}

const mapStateToProps = state => {
  const { cart } = state;

  return {
    cart
  };
};

export const ConnectedCart = connect(mapStateToProps)(Cart);

Cart.propTypes = {
  cart: PropTypes.object,
  dispatch: PropTypes.func.isRequired,
  setScreen: PropTypes.func.isRequired
};
