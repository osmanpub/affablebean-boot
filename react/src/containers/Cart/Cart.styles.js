import styled from "styled-components";

export const EmptyCart = styled.div`
  display: ${cart => (cart.numberOfItems === 0 ? "block" : "none")};
`;
