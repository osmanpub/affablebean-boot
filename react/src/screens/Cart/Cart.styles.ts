import styled from "styled-components";
import { Cart } from "../../interfaces/cart";

export const ActionBar = styled.div`
  margin: 30px;
  width: 750px;
  text-align: center;
`;

export const CartTable = styled.table`
  width: 750px;
`;

export const CartTableTd = styled.td`
  width: 145px;
  height: 90px;
`;

export const ShoppingCart = styled.div`
  display: ${(cart: Cart) => (cart.numberOfItems > 0 ? "block" : "none")};
`;

export const Subtotal = styled.h4`
  margin: 40px 0 20px 430px;
`;
