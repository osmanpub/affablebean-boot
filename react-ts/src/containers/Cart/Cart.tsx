import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import CartItem from "../../components/CartItem";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { ActionBar, CartTable, ShoppingCart, Subtotal } from "./Cart.styles";
import { clearCart } from "../../actions";
import { Props } from "../../interfaces/props";
import { CartItem as ICartItem } from "../../interfaces/cart";

export class Cart extends Component<Props> {
  componentDidUpdate(prevProps: Props) {
    const { cart, dispatch, match } = this.props;
    const { params } = match;

    if (cart.numberOfItems > 0 && params.clear === "true") {
      dispatch(clearCart());
    }
  }

  render() {
    const { cart, dispatch, match } = this.props;
    const { numberOfItems } = cart;

    const items = cart.items.map((item: ICartItem, index) => (
      <CartItem key={index} dispatch={dispatch} index={index} item={item} />
    ));

    const clearCart =
      numberOfItems === 0 ? (
        ""
      ) : (
        <Link className="bubble hMargin" to={"/viewCart/true"}>
          clear cart
        </Link>
      );

    return (
      <div>
        <Header cart={cart} url={match.url} />
        <div className="singleColumn">
          {numberOfItems === 0
            ? "Your shopping cart is empty"
            : "Your shopping cart contains " + numberOfItems + " items"}
          <ActionBar>
            {clearCart}
            <Link className="bubble hMargin" to={"/"}>
              continue shopping
            </Link>
          </ActionBar>
          <ShoppingCart {...cart}>
            <Subtotal>
              subtotal &euro;
              {cart.subtotal.toFixed(2)}
            </Subtotal>
            <CartTable>
              <tbody>
                <tr className="header">
                  <th>product</th>
                  <th>name</th>
                  <th>price</th>
                  <th>quantity</th>
                </tr>
                {items}
              </tbody>
            </CartTable>
          </ShoppingCart>
        </div>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = (state: Props) => {
  const { cart } = state;

  return {
    cart
  };
};

export default connect(mapStateToProps)(Cart);
