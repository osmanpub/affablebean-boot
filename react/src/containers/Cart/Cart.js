import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import CartItem from "../../components/CartItem";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { ActionBar, CartTable, ShoppingCart, Subtotal } from "./Cart.styles";
import { clearCart } from "../../actions";

export class Cart extends Component {
  componentDidUpdate(prevProps) {
    const { cart, dispatch, match } = this.props;
    const { params } = match;

    if (cart.numberOfItems > 0 && params.clear === "true") {
      dispatch(clearCart());
    }
  }

  render() {
    const { cart, dispatch, match } = this.props;
    const { numberOfItems } = cart;

    const items = cart.items.map((item, index) => (
      <CartItem
        key={index}
        cart={cart}
        dispatch={dispatch}
        index={index}
        item={item}
      />
    ));

    const clearCart =
      numberOfItems === 0 ? (
        ""
      ) : (
        <Link className="bubble hMargin" to={"/viewCart/true"}>
          clear cart
        </Link>
      );

    return (
      <div>
        <Header cart={cart} url={match.url} />
        <div className="singleColumn">
          {numberOfItems === 0
            ? "Your shopping cart is empty"
            : "Your shopping cart contains " + numberOfItems + " items"}
          <ActionBar>
            {clearCart}
            <Link className="bubble hMargin" to={"/"}>
              continue shopping
            </Link>
          </ActionBar>
          <ShoppingCart {...cart}>
            <Subtotal>
              subtotal &euro;
              {cart.subtotal.toFixed(2)}
            </Subtotal>
            <CartTable>
              <tbody>
                <tr className="header">
                  <th>product</th>
                  <th>name</th>
                  <th>price</th>
                  <th>quantity</th>
                </tr>
                {items}
              </tbody>
            </CartTable>
          </ShoppingCart>
        </div>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { cart } = state;

  return {
    cart
  };
};

export default connect(mapStateToProps)(Cart);

Cart.propTypes = {
  cart: PropTypes.object,
  dispatch: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired,
  url: PropTypes.string
};
