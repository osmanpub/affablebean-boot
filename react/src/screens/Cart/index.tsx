import React, { useEffect } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import CartItem from "../../components/CartItem";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Cart as CartState } from "../../interfaces/cart";
import { Match } from "../../interfaces/router";
import { RootState } from "../../redux";
import { clearCart } from "../../redux/cart";
import { ActionBar, CartTable, ShoppingCart, Subtotal } from "./Cart.styles";

type Props = {
  cart: CartState;
  clearCart: Function;
  match: Match;
};

function Cart(props: Props) {
  const { cart, clearCart, match } = props;
  const { numberOfItems } = cart;

  useEffect(() => {
    const { params } = match;

    if (cart.numberOfItems > 0 && params.clear === "true") {
      clearCart();
    }
  }, [cart.numberOfItems, clearCart, match]);

  const items = cart.items.map((item, index) => (
    <CartItem key={index} index={index} item={item} />
  ));

  const clear =
    numberOfItems === 0 ? (
      ""
    ) : (
      <Link
        className="bubble hMargin"
        to={"/viewCart/true"}
        data-cy="clear-cart"
      >
        clear cart
      </Link>
    );

  return (
    <div>
      <Header url={match.url} />
      <div className="singleColumn" data-cy="cart-info">
        {numberOfItems === 0
          ? "Your shopping cart is empty"
          : "Your shopping cart contains " + numberOfItems + " items"}
        <ActionBar>
          {clear}
          <Link className="bubble hMargin" to={"/"} data-cy="continue-shopping">
            continue shopping
          </Link>
        </ActionBar>
        <ShoppingCart {...cart}>
          <Subtotal>
            subtotal &euro;
            {cart.subtotal.toFixed(2)}
          </Subtotal>
          <CartTable>
            <tbody>
              <tr className="header">
                <th>product</th>
                <th>name</th>
                <th>price</th>
                <th>quantity</th>
              </tr>
              {items}
            </tbody>
          </CartTable>
        </ShoppingCart>
      </div>
      <Footer />
    </div>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart
});

const mapDispatchToProps = {
  clearCart
};

export default connect(mapStateToProps, mapDispatchToProps)(Cart);
