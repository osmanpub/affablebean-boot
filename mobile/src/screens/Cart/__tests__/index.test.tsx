import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Cart from '..';

const mockStore = configureStore([]);

const initialState = {
  cart: {
    didInvalidate: false,
    isFetching: false,
    items: [
      {
        product: {
          id: 1,
          description: 'semi skimmed (1L)',
          name: 'milk',
          price: 1.7,
        },
        quantity: 1,
        total: 1.7,
      },
    ],
    numberOfItems: 1,
    subtotal: 1.7,
  },
};

const store = mockStore(initialState);

describe('<Cart />', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <Cart setScreen={jest.fn()} />
        </Provider>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
