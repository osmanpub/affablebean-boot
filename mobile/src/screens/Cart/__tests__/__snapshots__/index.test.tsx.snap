// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> renders correctly 1`] = `
<ScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/components/Header/logo.jpg",
          }
        }
        style={
          Object {
            "margin": 16,
          }
        }
        testID="logo"
      />
    </TouchableWithoutFeedback>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingBottom": 16,
          }
        }
      >
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[Function]}
          underlayColor="black"
        >
          <Text
            style={
              Object {
                "borderColor": "blue",
                "borderRadius": 32,
                "borderWidth": 1,
                "color": "blue",
                "fontWeight": "bold",
                "marginHorizontal": 8,
                "padding": 12,
              }
            }
            testID="clear-cart"
          >
            clear cart
          </Text>
        </TouchableHighlight>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Text
            style={
              Object {
                "borderColor": "blue",
                "borderRadius": 32,
                "borderWidth": 1,
                "color": "blue",
                "fontWeight": "bold",
                "marginHorizontal": 8,
                "padding": 12,
              }
            }
            testID="checkout"
          >
            checkout
          </Text>
        </TouchableWithoutFeedback>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 16,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/components/Header/cart.gif",
            }
          }
        />
        <Text
          testID="cart-items"
        >
           
          1
           items
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 24,
        "paddingLeft": 32,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/img/products/milk.png",
        }
      }
      testID="image-milk"
    />
    <View
      style={
        Object {
          "flexDirection": "column",
          "paddingLeft": 24,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
        testID="name-milk"
      >
        milk
      </Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        € 
        1.70
      </Text>
      <Text
        style={
          Object {
            "paddingBottom": 8,
          }
        }
        testID="price-milk"
      >
        € 
        1.70
      </Text>
      <TextInput
        allowFontScaling={true}
        name="quantity"
        onChange={[Function]}
        placeholder="Enter quantity"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "marginBottom": 8,
          }
        }
        underlineColorAndroid="transparent"
        value="1"
      />
      <Button
        onPress={[Function]}
        testID="submit-milk"
        title="update"
      />
    </View>
  </View>
</ScrollView>
`;
